### PR TITLE
[MNG-7642] Restore compatibility with flatten-maven-plugin

### PR DESCRIPTION
**Pull request by gnodet**
Revert "[MNG-7505] Remove ReportingConverter (#906)"
This reverts commit adf89ef63ffca07e490f5cc550f9411bad1a7bd6.

Following this checklist to help us incorporate your
contribution quickly and easily:

 - [ ] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/MNG) filed
       for the change (usually before you start working on it).  Trivial changes like typos do not
       require a JIRA issue. Your pull request should address just this issue, without
       pulling in other changes.
 - [ ] Each commit in the pull request should have a meaningful subject line and body.
 - [ ] Format the pull request title like `[MNG-XXX] SUMMARY`,
       where you replace `MNG-XXX` and `SUMMARY` with the appropriate JIRA issue.
 - [ ] Also format the first line of the commit message like `[MNG-XXX] SUMMARY`.
       Best practice is to use the JIRA issue title in both the pull request title and in the first line of the commit message.
 - [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [ ] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will
       be performed on your pull request automatically.
 - [ ] You have run the [Core IT][core-its] successfully.

If your pull request is about ~20 lines of code you don't need to sign an
[Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf) if you are unsure
please ask on the developers list.

To make clear that you license your contribution under
the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
you have to acknowledge this by using the following check-box.

 - [ ] I hereby declare this contribution to be licenced under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)

 - [ ] In any other case, please file an [Apache Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

[core-its]: https://maven.apache.org/core-its/core-it-suite/

### CODE REVIEW

**Review by slawekjaranowski — APPROVED** [review]

File: maven-model-builder/src/main/java/org/apache/maven/model/building/DefaultModelBuilder.java [hunk 1]
```diff
@@ -625,6 +626,15 @@ public DefaultModelBuilder setProfileActivationFilePathInterpolator(
                 versionProcessor);
     }
 
+    /**
+     * @deprecated since Maven 4
+     * @see DefaultModelBuilderFactory#setReportingConverter(ReportingConverter)
```
Comment (pzygielo) [inline maven-model-builder/src/main/java/org/apache/maven/model/building/DefaultModelBuilder.java:631]: There is nothing to be seen there.

**Comment by pzygielo** [comment]
Not sure about _Restore compatibility_.
This just restores as no-op methods. If flatten called them - didn't it do it for some purpose?
Will flatten behave as expected or just not throw API incompatibility exception and silently do who-knows-what? Perhaps flatten should be updated instead?

I admit - I have not reviewed this PR/MNG-7642 in detail, so it might all be fine. No need to reply to my comment.

**Comment by gnodet** [comment]
> Not sure about _Restore compatibility_. This just restores as no-op methods. If flatten called them - didn't it do it for some purpose? Will flatten behave as expected or just not throw API incompatibility exception and silently do who-knows-what? Perhaps flatten should be updated instead?
> 
> I admit - I have not reviewed this PR/[MNG-7642](https://issues.apache.org/jira/browse/MNG-7642) in detail, so it might all be fine. No need to reply to my comment.

Those classes are actually not used.  But the flatten maven plugin needs to be injected with one instance of `ReportingConverter`, see https://github.com/mojohaus/flatten-maven-plugin/blob/master/src/main/java/org/codehaus/mojo/flatten/ModelBuilderThreadSafetyWorkaround.java#L107
It will simply pass it to the `ModelBuilder` in the loop method.

The flatten plugin should also be adapted to support the removal of those classes, so that we can get rid of those in 4.1 for example.

File: maven-model-builder/src/main/java/org/apache/maven/model/building/DefaultModelBuilder.java (hunk 1 above)
Comment (gnodet) [reply maven-model-builder/src/main/java/org/apache/maven/model/building/DefaultModelBuilder.java:631]: All the `setXxx` methods have the same javadoc.  The deprecation is not something new in this case.

**Comment by pzygielo** [comment]
So it will be passed in
https://github.com/apache/maven/blob/aab448546326da99fd8f2fc5d15ff08dd79bae8e/src/main/java/org/codehaus/mojo/flatten/ModelBuilderThreadSafetyWorkaround.java#L131
just to be ignored?

**Comment by pzygielo** [comment]
I think I can see that now.
Thanks @gnodet.

**Comment by gnodet** [comment]
> I think I can see that now.
> Thanks @gnodet.

Yes, it's really just for binary compatibility with the existing stuff.

File: maven-model-builder/src/main/java/org/apache/maven/model/plugin/DefaultReportingConverter.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+import org.apache.maven.model.Model;
+import org.apache.maven.model.building.ModelBuildingRequest;
+import org.apache.maven.model.building.ModelProblemCollector;
+
+/**
+ * Handles conversion of the <code>&lt;reporting&gt;</code> section into the configuration of Maven Site Plugin 3.x,
+ * i.e. <code>reportPlugins</code> and <code>outputDirectory</code> parameters.
+ *
+ * @author Benjamin Bentmann
+ * @deprecated since maven 4.0, this class is not used anymore and is only here for compatibility
```
Comment (michael-o) [inline maven-model-builder/src/main/java/org/apache/maven/model/plugin/DefaultReportingConverter.java:33]: Don't should say that this is a no-op

**Review by michael-o — CHANGES_REQUESTED** [review]

**Review by michael-o — APPROVED** [review]

**Comment by jira-importer** [comment]
Resolve #8811

**Pull request merged**